window/src/lib.rs

```diff
@@ -294,6 +294,14 @@ pub trait WindowOps {
     /// Set some text in the clipboard
     fn set_clipboard(&self, clipboard: Clipboard, text: String);
 
+    /// Set window level (macos only!)
+    #[cfg(target_os = "macos")]
+    fn set_level(&self, level: WindowLevel);
+
+    /// Get window level (macos only!)
+    #[cfg(target_os = "macos")]
+    fn level(&self) -> Future<WindowLevel>;
```

Comment:
The concept of an "always-on-top" window is not specific to macOS, even though this PR only implements it there.

The interface in this module should allow for things to work on all platforms, even if the implementation isn't present there yet.

My suggestion is to draw some inspiration from the `winit` crate which has this [WindowLevel](https://docs.rs/winit/latest/winit/window/enum.WindowLevel.html) enum:

```rust
pub enum WindowLevel {
    AlwaysOnBottom,
    Normal,
    AlwaysOnTop,
}
```

I think we could have essentially the same enum here in this module, and define a `set_window_level` method for setting it.  I think `set_level` is not specific or descriptive enough as a name.

I'm not convinced that we need a method to retrieve the window level; winit doesn't have one, and avoiding adding another `Future` thing here simplifies the implementation.

```suggestion
    /// Set window level. Depending on the environment, and user preference,
    /// this may have no effect.
    fn set_window_level(&self, level: WindowLevel) {
      // Default is NOP
    }
```